packages/nx-plugin/src/schematics/builder/schema.json

```diff
@@ -0,0 +1,38 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "id": "NxPluginBuilder",
+  "title": "Create a Builder for an Nx Plugin",
+  "type": "object",
+  "examples": [
+    {
+      "command": "g builder my-builder --project=my-plugin",
+      "description": "Generate libs/my-plugin/src/builders/my-builder"
+    }
+  ],
+  "properties": {
+    "project": {
+      "type": "string",
+      "description": "The name of the project.",
+      "alias": "p",
+      "$default": {
+        "$source": "projectName"
```

Comment:
Is this working for you in a nx cli workspace?

Reply:
Is which part working? The `$default` stuff for this parameter?

Reply:
Yes with the angular cli the schematic gets with this code automatically the projectName (usually the default one which is defined). so if you call your schematic ng g schematic ... without --project it should work. I have the same with my plugin but I have to give him every time the project via --project. Which makes "$default" -> "$source" obsolete.

Can you try it for you if it is working? It might be a bug in nx

Reply:
@MitkoTschimev the `projectName` is a "smart default provider" provided by the Angular CLI to register keywords for the JSON schema validator. You can see them add `projectName` here:
https://github.com/angular/angular-cli/blob/35ca3901acbc867363272aa4a4ab06166f8da7b1/packages/angular/cli/models/schematic-command.ts#L318

If you're using the `nx` command by itself, the `projectName` default isn't registered. So you could consider this a bug. 

The place to add this to make sure that `nx` has support for it, would be in the `tao` package here:
https://github.com/nrwl/nx/blob/8caff7fcde671636f7c11295bcba3ddc1e44bf32/packages/tao/src/commands/generate.ts#L166-L172